ehitchhiking-ui/src/app/shared/components/trip/trip.component.ts

```diff
@@ -16,4 +19,15 @@ export class TripComponent implements OnInit {
 	makeFavorite() {
 		this.trip.isFavorite = !this.trip.isFavorite;
 	}
+
+  toggleRating(){
+	  this.isRating = !this.isRating;
+  }
+
+  rateTrip(clickObj: StarClickMeta): void {
+    if (!!this.trip) {
```

Comment:
early return is required here

if (!this.trip) {
return;
}

this.trip.rating = clickObj.rating;
 this.toggleRating();

Reply:
fixed